core/trie/snap_support.go

```diff
@@ -0,0 +1,168 @@
+package trie
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+func (t *Trie) IterateAndGenerateProof(startValue *felt.Felt, consumer func(key, value *felt.Felt) (bool, error),
```

Comment:
This function just generates the proofs in a given scenario right, it doesn't really iterate through anything?